tools/engine_tool/lib/src/commands/run_command.dart

```diff
@@ -0,0 +1,141 @@
+      defaultsTo: '',
+      help:
+          'Specify the build config to use for the target build (usually auto detected)',
+      allowed: <String>[
+        for (final Build build in runnableBuilds(environment, configs))
+          build.name,
+      ],
+      allowedHelp: <String, String>{
+        for (final Build build in runnableBuilds(environment, configs))
+          build.name: build.gn.join(' '),
+      },
+    );
+  }
+
+  /// List of compatible builds.
+  late final List<Build> builds;
+
+  @override
+  String get name => 'run';
+
+  @override
+  String get description => 'Invokes flutter run with a local engine build\n'
+      'All arguments after -- are forwarded to flutter run, e.g.: '
+      'et run -- --profile\n'
+      'See `flutter run --help` for a listing';
+
+  Build? _lookup(String configName) {
+    return builds.where((Build build) => build.name == configName).firstOrNull;
+  }
+
+  Build? _findHostBuild(Build? targetBuild) {
+    if (targetBuild == null) {
+      return null;
+    }
+
+    // TODO(johnmccutchan): This is brittle, it would be better if we encoded
+    // the host config name in the target config.
+    final String name = targetBuild.name;
+    if (name.contains('_debug')) {
+      return _lookup('host_debug');
+    } else if (name.contains('_profile')) {
+      return _lookup('host_profile');
+    } else if (name.contains('_release')) {
+      return _lookup('host_release');
+    }
+    return null;
+  }
+
+  String _selectTargetConfig() {
+    final String configName = argResults![configFlag] as String;
+    if (configName.isNotEmpty) {
+      return configName;
+    }
+    // TODO(johnmccutchan): We need a way to invoke flutter tool and be told
+    // which OS and CPU architecture the selected device requires, for now
+    // use some hard coded values:
+    const String targetOS = 'android';
+    const String cpuArch = 'arm64';
+
+    // Sniff the build mode from the args that will be passed to flutter run.
+    String mode = 'debug';
+    if (argResults!.rest.contains('--profile')) {
+      mode = 'profile';
+    } else if (argResults!.rest.contains('--release')) {
+      mode = 'release';
+    }
+
+    return '${targetOS}_${mode}_$cpuArch';
+  }
+
+  @override
+  Future<int> run() async {
+    final String configName = _selectTargetConfig();
+    final Build? build = _lookup(configName);
+    final Build? hostBuild = _findHostBuild(build);
+    if (build == null) {
+      environment.logger.error('Could not find build $configName');
+      return 1;
+    }
+    if (hostBuild == null) {
+      environment.logger.error('Could not find host build for $configName');
+      return 1;
+    }
+
+    // First build the host.
+    int? r = await runner!.run(<String>['build', '-c', hostBuild.name]);
+    if (r != null && r != 0) {
+      return r;
+    }
+
+    // Now build the target.
+    r = await runner!.run(<String>['build', '-c', build.name]);
+    if (r != null && r != 0) {
+      return r;
+    }
```

Comment:
We shouldn't run the target build if the host and target builds are the same

Reply:
Fixed.